src/client/pythonEnvironments/legacyIOC.ts

```diff
@@ -204,30 +208,35 @@ class ComponentAdapter implements IComponentAdapter {
         if (!(await this.isEnabled())) {
             return undefined;
         }
-        const env = await this.api.resolveEnv(interpreterPath);
-        if (env === undefined) {
-            return undefined;
-        }
-        return env.kind === PythonEnvKind.Conda;
+        // While `ComponentAdapter` represents how the component would be used in the rest of the
+        // extension, we cheat here for the sake of performance.  This is not a problem because when
+        // we start using the component's public API directly we will be dealing with `PythonEnvInfo`
+        // instead of just `pythonPath`.
+        return isCondaEnvironment(interpreterPath);
     }
 
     // A result of `undefined` means "Fall back to the old code!"
     public async getCondaEnvironment(interpreterPath: string): Promise<CondaEnvironmentInfo | undefined> {
         if (!(await this.isEnabled())) {
             return undefined;
         }
-        const env = await this.api.resolveEnv(interpreterPath);
-        if (env === undefined) {
-            return undefined;
-        }
-        if (env.kind !== PythonEnvKind.Conda) {
+        // While `ComponentAdapter` represents how the component would be used in the rest of the
+        // extension, we cheat here for the sake of performance.  This is not a problem because when
+        // we start using the component's public API directly we will be dealing with `PythonEnvInfo`
+        // instead of just `pythonPath`.
+        //
+        // Also, for Conda we assume we don't set name for environments if they're prefix conda
+        // environments, similarly we don't have 'path' set if they're non-prefix conda environments.
+        // tslint:disable-next-line:no-suspicious-comment
+        // TODO: Expose these two properties via a method in the Conda locator on a temporary basis. This method
+        // will go away eventually, and so exposing those properties will no longer be needed. Until then we only
+        // return location.
+        const location = getEnvironmentDirFromPath(interpreterPath);
+        if (!(await isCondaEnvironment(interpreterPath))) {
             return undefined;
         }
```

Comment:
Two things then: call `isCondaEnvironment()` first and clarify the comment.

```suggestion
        if (!(await isCondaEnvironment(interpreterPath))) {
            return undefined;
        }
        // We don't have a helper function yet to give us a conda env's name (if it has one).  So for
        // now we always set `path` (and never `name`).  Once we have such a helper we will use it.
        const location = getEnvironmentDirFromPath(interpreterPath);
```